src/hooks/useStore.js

```diff
@@ -0,0 +1,35 @@
+import create from 'zustand';
+import {persist} from 'zustand/middleware';
+
+const shoppingItemsStore = set => ({
+  shoppingItems: [],
+  addShoppingItem: item => {
+    set(state => ({
+      shoppingItems: [item, ...state.shoppingItems],
+    }));
+  },
+  removeShoppingItem: itemId => {
+    set(state => ({
+      shoppingItems: state.shoppingItems.filter(item => item.id !== itemId),
+    }));
+  },
+
+  plusItemQuantity: itemId => {
+    set(state => ({
+      shoppingItems: state.shoppingItems.map(item =>
+        item.id === itemId ? {...item, quantity: item.quantity + 1} : item
+      ),
+    }));
+  },
+  minusItemQuantity: itemId => {
+    set(state => ({
+      shoppingItems: state.shoppingItems.map(item =>
+        item.id === itemId ? {...item, quantity: item.quantity > 1 ? item.quantity - 1 : item.quantity} : item
```

Comment:
So the quantity can never be reduced to 0?

Reply:
0 means delete, I'm thinking add a message here to notify the users, that quantity can't be less than 1.

Reply:
That is a good idea :)